src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs

```diff
@@ -742,27 +742,25 @@ private void ProcessMessageProc(object state)
         }
 
         private const string GUIDTAG = "PSGuid='";
-        private const int GUID_STR_LEN = 36;        // GUID string: 32 digits plus 4 dashes
 
-        private Guid GetMessageGuid(string data)
+        private bool ContainsMessageGuid(ReadOnlySpan<char> data)
         {
-            // Perform quick scan for data packet for a GUID, ignoring any errors.
-            var iTag = data.IndexOf(GUIDTAG, StringComparison.OrdinalIgnoreCase);
-            if (iTag > -1)
+            try
             {
-                try
-                {
-                    var psGuidString = data.Substring(iTag + GUIDTAG.Length, GUID_STR_LEN);
-                    return new Guid(psGuidString);
-                }
-                catch 
+                // Perform quick scan for data packet for a GUID, ignoring any errors.
+                var iTag = data.IndexOf(GUIDTAG, StringComparison.OrdinalIgnoreCase);
+                if (iTag > -1)
                 {
-                    // Ignore any malformed packet errors here and return an empty Guid.
-                    // Packet errors will be reported later during message processing.
+                    var psGuidString = data.Slice(iTag + GUIDTAG.Length);
+                    return Guid.TryParse(psGuidString, out _);
```

Comment:
Using Span.Slice() is a good optimization.  But this code is actually incorrect as is.  All PSRP packets contain a Guid.  But a session message guid is always an empty guid (which is why the original code compares to an empty Guid).  A non-empty guid means the PSRP packet is a command message.  This method is used to route the message to the appropriate processing thread.  To fix this, please make the following changes;

```
private bool IsCommandMessage(ReadOnlySpan<char> data)
{
    try
    {
        // Perform quick scan for data packet GUID.
        var iTag = data.Index(GUIDTAG, StringComparison.OrdinalIgnoreCase);
        if (iTag > -1)
        {
            // An empty GUID value means this is a session message, otherwise it is a command message.
            var psGuidString = data.Slice(iTag + GUIDTag.Length, GUID_STR_LEN);
            if (Guid.TryParse(psGuidString out Guid psGuid))
            {
                return !Guid.Equals(psGuid, Guid.Empty);
            }
        }
    }
    catch
    {
        // The method should never throw.
    }

    // Missing Guid means an invalid packet.  Continue processing as session message.
    return false;
}
```

Reply:
Ah, I did know. Thanks!

Do we know that GUID always present? If yes we could only search the const:
```suggestion
        private const string GUIDTAG = "PSGuid=00000000-0000-0000-0000-000000000000'";
```

Reply:
Yes, good point.  We can simply search for an empty PSGuid element.  All valid packets will have a `PSGuid` element, and if invalid we just treat as a session message.

Reply:
Done.

Reply:
I think we can make it even simpler.  I now agree that the method won't throw so we can remove the try/catch.  Also empty Guid equates to session message so we should name the method correctly.  I was thinking:
```
private const string SESSIONGUID = "PSGuid='00000000-0000-0000-0000-000000000000'";
private bool IsSessionMessage(string data)
{
    // Perform quick scan for an empty GUID element, which identifies a session message.
    return data.IndexOf(SESSIONGUID, StringComparison.OrdinalIgnoreCase) > -1;
}
```

Reply:
Since this is only used for routing, any *invalid* packet can be routed to command thread ... it really shouldn't matter.

Reply:
Done.